src/components/PostInfo/PostInfo.tsx

```diff
@@ -1,5 +1,36 @@
 import React from 'react';
+import { CommentList } from '../CommentList';
+import { UserInfo } from '../UserInfo';
+import { PostInfoType } from '../../type/post';
+import './PostInfo.scss';
 
-export const PostInfo: React.FC = () => (
-  <>Put the post here</>
+export const PostInfo: React.FC<PostInfoType> = ({
+  post: {
+    id,
+    title,
+    body,
+    user,
+    comments,
+  },
+}) => (
+  <div className="PostInfo" key={id}>
+    <div className="PostInfo__header">
+      <h3 className="PostInfo__title">{title}</h3>
+
+      {user && <UserInfo user={user} />}
+    </div>
+
+    <p className="PostInfo__body">
+      {body}
+    </p>
+
+    <hr />
+
+    {comments && comments.length > 0 ? (
+      <CommentList comments={comments} />
+    )
+      : (
+        <b data-cy="NoCommentsMessage">No comments yet</b>
+      )}
```

Comment:
```suggestion
    {comments.length > 0 ? (
      <CommentList comments={comments} />
    ) : (
      <b data-cy="NoCommentsMessage">No comments yet</b>
    )}
```
keep the value of the `comments` variable always array type and remove the additional check, also this formatting will look a bit better